lib/types/protocol-version.js

```diff
@@ -127,6 +135,73 @@ const protocolVersion = {
       return 0;
     }
     return version - 1;
+  },
+
+  /**
+   * Computes the highest supported protocol version collectively by the given hosts.
+   *
+   * Considers the cassandra_version of the input hosts to determine what protocol versions
+   * are supported and uses the highest common protocol version amont them.
+   *
+   * If hosts >= C* 3.0 are detected, any hosts older than C* 2.1 will not be considered
+   * as those cannot be connected to.  In general this will not be a problem as C* does
+   * not support clusters with nodes that have versions that are more than one major
+   * version away from each other.
+   * @param {Connection} connection Connection hosts were discovered from.
+   * @param {Array.<Host>} hosts The hosts to determine highest protocol version from.
+   * @return {Number} Highest supported protocol version among hosts.
+   */
+  getHighestCommon: function(connection, hosts) {
+    const log = connection.log ? connection.log.bind(connection) : utils.noop;
+    let maxVersion = connection.protocolVersion;
```

Comment:
Possible issue here, though probably not a valid use case:

control connection is to a host that supports V2 at maximum (i.e. C* 2.0, there are other hosts that require at least V3 (i.e. C* 3.0).

In this case this method will return V2, and hosts that require V3 can't be connected to.  I think the best we can do here is log this scenario, as we do at line 198 when we detect hosts that don't support V3.

Reply:
I added some additional logging for this case . Technically it's not valid since you shouldn't be able to have a cluster with nodes separated by more than one major version, so i log something at error.

Reply:
I think logging an error is enough, no sane cluster should have this issue.